pkg/kubelet/kuberuntime/kuberuntime_image.go

```diff
@@ -135,14 +137,40 @@ func (m *kubeGenericRuntimeManager) RemoveImage(image kubecontainer.ImageSpec) e
 // this is a known issue, and we'll address this by getting imagefs stats directly from CRI.
 // TODO: Get imagefs stats directly from CRI.
 func (m *kubeGenericRuntimeManager) ImageStats() (*kubecontainer.ImageStats, error) {
-	allImages, err := m.imageService.ListImages(nil)
+	imageFSInfo, err := m.imageService.ImageFsInfo()
 	if err != nil {
-		glog.Errorf("ListImages failed: %v", err)
+		glog.Errorf("ImageFsInfo failed: %v", err)
 		return nil, err
 	}
-	stats := &kubecontainer.ImageStats{}
-	for _, img := range allImages {
-		stats.TotalStorageBytes += img.Size_
+
+	return &kubecontainer.ImageStats{
+		TotalStorageBytes: imageFSInfo.UsedBytes.GetValue(),
+	}, nil
+}
+
+// ImageFsInfo gets filesytem information of images.
+func (m *kubeGenericRuntimeManager) ImageFsInfo() (cadvisorapiv2.FsInfo, error) {
```

Comment:
Why is the `kubeGenericRuntimeManager` constructing a `cadvisorapiv2.FsInfo`? I think the dependencies are a bit scrambled. At the very least, you could clone the API into Kubernetes.